### PR TITLE
Rename objects returned by server

### PR DESCRIPTION
**Pull request by julienw**

### CODE REVIEW

File: app/js/views/reminders.jsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default class Reminders extends React.Component {
     const reminders = this.state.reminders;
     reminders.push({
       id: reminders.length,
-      recipient: reminder.users,
```
Comment (julienw) [inline app/js/views/reminders.jsx:70]: I have a doubt here: what was "recipient" supposed to hold ? An array or a string ?

File: app/push-sw.js [hunk 2]
```diff
@@ -72,8 +72,18 @@ function notifyClient(obj) {
...
+  let title;
+  if (!obj.recipients.length) {
+    title = 'Reminder';
+  } else if (obj.recipients.length === 1) {
+    title = `Reminder for ${obj.recipients[0]}`;
+  } else {
+    const firstRecipients = obj.recipients.slice(0, -1).join(', ');
+    const lastRecipient = obj.recipients[obj.recipients.length - 1];
+    title = `Reminder for ${firstRecipients} and ${lastRecipient}`;
+  }
```
Comment (julienw) [inline app/push-sw.js:84]: quite complex logic to have a nice title.

**Comment by julienw** [comment]
r? @JohanLorenzo 
r? @gmarty 

I rebased https://github.com/fxbox/calendar/pull/17 and added 3 commits on top of it.

File: app/push-sw.js (hunk 2 above)
Comment (JohanLorenzo) [inline app/push-sw.js:84]: Super nit picking: Unlike French, English sentences have a comma before the "and" in a list of 3 or more elements. For instance:

> I have apples, bananas, and oranges.

http://www.grammarbook.com/punctuation/commas.asp (Rule 1)

**Comment by JohanLorenzo** [comment]
Looks good to me

File: app/push-sw.js [hunk 3]
```diff
@@ -72,8 +72,18 @@ function notifyClient(obj) {
...
-  const body = obj.message;
+  let title;
+  if (!obj.recipients.length) {
+    title = 'Reminder';
+  } else if (obj.recipients.length === 1) {
+    title = `Reminder for ${obj.recipients[0]}`;
+  } else {
+    const firstRecipients = obj.recipients.slice(0, -1).join(', ');
+    const lastRecipient = obj.recipients[obj.recipients.length - 1];
+    title = `Reminder for ${firstRecipients}, and ${lastRecipient}`;
```
Comment (gmarty) [inline app/push-sw.js]: I saw your comment about the Oxford comma. First of all it's not standard, some people use it, other don't. It's not an absolute rule. Second of all, it's only for list of 3 items or more. In this case, we'll still have the comma for 2 items. So I'd suggest to remove it for now.
In the future, I'm thinking of using something like https://github.com/twitter/twitter-cldr-js/blob/master/lib/assets/javascripts/twitter_cldr/en.js#L3188 to generate internationalised list of items.

File: app/js/views/reminders.jsx (hunk 1 above)
Comment (gmarty) [inline app/js/views/reminders.jsx:70]: It's supposed to be an array. Did you make the change on the server?

**Comment by gmarty** [comment]
r+ with a nit fixed.

**Comment by julienw** [comment]
I fixed the nit. We'll merge and deploy monday :)

**Comment by JohanLorenzo** [comment]
Time to deploy!

**Pull request merged**